pkg/syncer/storage/storage_pv_process.go

```diff
@@ -0,0 +1,111 @@
+
+	kcpcache "github.com/kcp-dev/apimachinery/pkg/cache"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/logging"
+	"github.com/kcp-dev/kcp/pkg/syncer/shared"
+)
+
+func (c *PersistentVolumeController) process(ctx context.Context, key string) error {
+	logger := klog.FromContext(ctx)
+
+	// Note "upstreamNamespace" will be empty since PersistentVolume is cluster-scoped
+	clusterName, upstreamNamespace, name, err := kcpcache.SplitMetaClusterNamespaceKey(key)
+	if err != nil {
+		logger.Error(err, "Invalid key", key)
+		return nil
+	}
+
+	logger = logger.WithValues(logging.WorkspaceKey, clusterName, logging.NameKey, name)
+
+	upstreamPVObject, err := c.getUpstreamPersistentVolume(clusterName, name)
+	if err != nil {
+		return fmt.Errorf("failed to get upstream PersistentVolume: %w", err)
+	}
+
+	upstreamPV, ok := upstreamPVObject.(*corev1.PersistentVolume)
+	if !ok {
+		return fmt.Errorf("failed to assert object to a PersistentVolume: %T", upstreamPVObject)
+	}
+
+	resourceState, ok := upstreamPV.GetLabels()[workloadv1alpha1.ClusterResourceStateLabelPrefix+c.syncTarget.key]
+	if ok && resourceState == string(workloadv1alpha1.ResourceStateUpsync) {
+		desiredNSLocator := shared.NewNamespaceLocator(clusterName, c.syncTarget.workspace, c.syncTarget.uid, c.syncTarget.name, upstreamNamespace)
+		downstreamPVObject, err := c.getDownstreamPersistentVolumeFromNamespaceLocator(desiredNSLocator)
+		if apierrors.IsNotFound(err) {
+			logger.V(4).Info("downstream persistent volume not found, ignoring key")
+			return nil
+		} else if err != nil {
+			logger.Error(err, "failed to get downstream persistent volume")
+			return nil
+		}
+
+		if downstreamPVObject == nil {
+			logger.Info("downstream persistent volume not found, ignoring key")
+			return nil
+		}
+
+		downstreamPV, ok := downstreamPVObject.(*corev1.PersistentVolume)
+		if !ok {
+			return fmt.Errorf("failed to assert object to a PersistentVolume: %T", downstreamPVObject)
+		}
+
+		downstreamPVCObject, err := c.getDownstreamPersistentVolumeClaim(downstreamPV.Spec.ClaimRef.Name, downstreamPV.Spec.ClaimRef.Namespace)
+		if err != nil {
+			return fmt.Errorf("failed to get downstream PersistentVolumeClaim: %w", err)
+		}
+
+		downstreamPVC, ok := downstreamPVCObject.(*corev1.PersistentVolumeClaim)
+		if !ok {
+			return fmt.Errorf("failed to assert object to a PersistentVolumeClaim: %T", downstreamPVCObject)
+		}
+
+		// Remove the internal.workload.kcp.dev/delaystatussyncing annotation
+		removeAnnotationFromPersistentVolumeClaim(downstreamPVC)
+
+		_, err = c.updateDownstreamPersistentVolumeClaim(ctx, downstreamPVC)
+		if err != nil {
+			return fmt.Errorf("failed to update downstream PersistentVolumeClaim: %w", err)
+		}
+
+		logger.V(1).Info("Removed the", DelayStatusSyncing, "annotation from downstream PersistentVolumeClaim", downstreamPVC.Name)
```

Comment:
nit: logging key-value keys should not contain spaces

Reply:
I see numerous instances in the code with loggers using spaces, I'm a bit confused by this rule. Or perhaps I misunderstood what you meant?

Reply:
All of them should stop - it's bad UX for selecting things. Consider the difficulty of using `jq` to select on the key `"annotation from downstream"`

Reply:
It's hard to build meaningful human-readable logs then, but anyways, thanks for clarifying.

Reply:
Structured log _keys_ are not for humans. Expect those to be used to generate e.g. columns in tables, or for selection, etc. _Values_ can be whatever you'd like :)